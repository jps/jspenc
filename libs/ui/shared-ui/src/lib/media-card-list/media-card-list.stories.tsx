import { Meta } from '@storybook/react';
import { BaseProvider } from '../..';
import { MediaCardList } from './media-card-list';
import { MediaCard } from '../media-card/media-card';

export default {
  component: MediaCardList,
  title: 'MediaCardList',
} as Meta;

export const TestCard = () =>
  <BaseProvider>
    <MediaCardList title='Latest Posts'>
      <MediaCard href="#" headline='Hello World'>something something something else</MediaCard>
    </MediaCardList>
  </BaseProvider>;