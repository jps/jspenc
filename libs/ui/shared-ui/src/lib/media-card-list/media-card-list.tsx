import { Block, TitleBar, Stack, styled } from 'newskit';

export interface MediaCardListProps {
  title: string;
  children: Exclude<React.ReactNode, 'undefined'>;
}

const Section = styled.section`
  width: 100%;
`;

export const MediaCardList = ({ title, children }: MediaCardListProps) => {
  const cards = Array.isArray(children) ? children : [children];
  return (
    <Section>
      {/* This block override has been added due to a bug in the title bar */}
      <Block marginBlockEnd="space060">
        <TitleBar headingAs="h2">{title}</TitleBar>
      </Block>
      <Stack flow="vertical-left" spaceInline="space060" list>
        {cards}
      </Stack>
    </Section>
  );
};
