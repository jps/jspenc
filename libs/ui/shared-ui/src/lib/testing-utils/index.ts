export * from './mock-window-matchMedia'
export * from './render-with-theme'