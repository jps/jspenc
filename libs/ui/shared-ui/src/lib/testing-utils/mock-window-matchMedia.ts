//Not yet implemented by JSDOM https://jestjs.io/docs/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
export const mockMatchMedia = () => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation(query => ({
        matches: false,
        media: query,
        onchange: null,
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  }
  