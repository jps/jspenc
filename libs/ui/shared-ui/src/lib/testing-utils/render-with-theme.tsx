import { render } from '@testing-library/react';
import { BaseProvider } from '../..';

export const renderWithTheme =
    (ui:React.ReactElement) => render(<BaseProvider>{ui}</BaseProvider>);