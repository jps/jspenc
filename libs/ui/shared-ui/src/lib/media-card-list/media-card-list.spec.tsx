import { renderWithTheme } from '../testing-utils';
import { MediaCardList } from './media-card-list';

const mediaCardListProps = [ {
    href: "https://jspenc.com",
    headline: "Jspenc",
    children : "lorem ipsum"
  }];

describe('MediaCardList', () => {
  it('should render successfully', () => {
    const { baseElement } = renderWithTheme(<MediaCardList title="Content" cards={mediaCardListProps} />);
    expect(baseElement).toMatchSnapshot();
  });
});
