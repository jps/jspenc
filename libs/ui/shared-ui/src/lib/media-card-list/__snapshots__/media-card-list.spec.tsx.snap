// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaCardList should render successfully 1`] = `
<body>
  <div>
    <section
      class="css-sxfrcw"
    >
      <div
        class="css-1r3hoet"
      >
        <div
          class="css-1qdltth"
        >
          <div
            class="css-1edp8de"
          >
            <section
              class="css-1eqp48b"
            >
              <h2
                class="nk-headline-heading css-1qhitfs"
              >
                Content
              </h2>
            </section>
          </div>
        </div>
      </div>
      <ul
        class="css-qtuc0p"
      >
        <li
          class="css-1cj9kmv"
        >
          <div>
            <div
              class="css-10n9oie"
            >
              <div
                class="css-1shocxe"
              >
                <div
                  class="css-1a2irou"
                >
                  <div
                    class="css-15bubm0"
                  >
                    <div
                      class="css-zodxql"
                    >
                      <a
                        class="nk-card-link css-1h8m54y"
                        href="https://jspenc.com"
                        target="_blank"
                      >
                        <section
                          class="css-4peoqn"
                        >
                          <h3
                            class="nk-headline-heading css-cjz1e9"
                          >
                            Jspenc
                          </h3>
                        </section>
                      </a>
                    </div>
                    <div
                      class="css-zodxql"
                    >
                      <p
                        class="css-1ffe19o"
                      >
                        lorem ipsum
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </li>
      </ul>
    </section>
  </div>
</body>
`;
